### PR TITLE
Segmentation contain filename

### LINKED ISSUE #317: Segmentation GeoJSON should contain human-readable filename
No-op, only for informational purpose.

### PR DESCRIPTION
**Pull request by GwydionJon**
This resolves #317

### CODE REVIEW

File: adaptivefiltering/execute.py [hunk 1]
```diff
@@ -119,8 +137,7 @@ def apply_adaptive_pipeline(
         )
 
         # Write the filter into the output directory
-        # TODO: Change this filename from hash to the saved filename once it is implemented
-        save_filter(filter, os.path.join(output_dir, "{hash}.json"))
+        save_filter(filter, os.path.join(output_dir, f"{filter.title}.json"))
```
Comment (dokempf) [inline adaptivefiltering/execute.py:140]: I think people will often use spaces as part of the title, but I consider it unfortunate to have files with spaces. Maybe something like `.lower().replace(" ", "_")` or similar should be used.

File: adaptivefiltering/execute.py [hunk 2]
```diff
@@ -66,14 +66,32 @@ def apply_adaptive_pipeline(
...
     # We add a file logging handler
     attach_file_logger(os.path.join(output_dir, "output.log"))
 
+    # if no spatial_refence is defined for the dataset it is tried to be extracted from the metadata
+    if dataset.spatial_reference is None:
+        from adaptivefiltering.pdal import PDALInMemoryDataSet
+
+        dataset.spatial_reference = PDALInMemoryDataSet.convert(
+            dataset
+        ).spatial_reference
```
Comment (dokempf) [inline adaptivefiltering/execute.py:83]: Is there some way of doing this without redundantly converting the dataset?

File: adaptivefiltering/execute.py [hunk 3]
```diff
@@ -66,14 +66,32 @@ def apply_adaptive_pipeline(
 
     # We decrease the logging level
     logger.setLevel(logging.INFO)
-
     # Ensure existence of output directory
+    # This is not written in the log file
```
Comment (dokempf) [inline adaptivefiltering/execute.py:70]: Should be?

**Review by dokempf — CHANGES_REQUESTED** [review]

File: adaptivefiltering/execute.py (hunk 2 above)
Comment (GwydionJon) [reply adaptivefiltering/execute.py:83]: only by forcing the user to manually set the spatial_reference information when loading a dataset.

File: adaptivefiltering/execute.py (hunk 3 above)
Comment (GwydionJon) [reply adaptivefiltering/execute.py:70]: Changed it to log, that the dict was created.

**Pull request merged**